src/citrine/resources/file_link.py

```diff
@@ -0,0 +1,111 @@
+            the file will be used. That is, the file at "/Users/me/diagram.pdf" will be uploaded
+            with the name "diagram.pdf".
+
+        Returns
+        -------
+        FileLink
+            The filename and url of the uploaded object.
+
+        """
+        if not os.path.isfile(file_path):
+            raise ValueError("No file at specified path {}".format(file_path))
+
+        if not dest_name:
+            # Use the file name as a default dest_name
+            dest_name = os.path.basename(file_path)
+
+        path = self._get_path() + "/uploads"
+        extension = os.path.splitext(file_path)[1]
+        mimeType = mimetypes.types_map[extension]
+        upload_json = {
+            'files': [
+                {
+                    'file_name': dest_name,
+                    'metadata': dict(),
+                    'mime_type': mimeType,
+                    'size': os.stat(file_path).st_size
+                }
+            ]
+        }
+        # POST request creates space in S3 for the file and returns AWS-related information
+        # (such as temporary credentials) that allow the file to be uploaded.
+        upload_data = self.session.post_resource(path=path, json=upload_json)
+
+        # Extract all relevant information from the POST response
+        try:
+            region_name = upload_data['s3_region']
+            aws_access_key_id = upload_data['temporary_credentials']['access_key_id']
+            aws_secret_access_key = upload_data['temporary_credentials']['secret_access_key']
+            aws_session_token = upload_data['temporary_credentials']['session_token']
+            bucket = upload_data['s3_bucket']
+            object_key = upload_data['uploads'][0]['s3_key']
+            upload_id = upload_data['uploads'][0]['upload_id']
+        except KeyError:
+            raise RuntimeError("Amazon S3 response is missing some fields: {}".format(upload_data))
+
+        s3_client = boto3_client('s3',
+                                 region_name=region_name,
+                                 aws_access_key_id=aws_access_key_id,
+                                 aws_secret_access_key=aws_secret_access_key,
+                                 aws_session_token=aws_session_token)
+        with open(file_path, 'rb') as f:
+            try:
+                upload_response = s3_client.put_object(Bucket=bucket, Key=object_key, Body=f)
+            except ClientError as e:
+                raise RuntimeError("Upload of file {} failed with the following "
+                                   "exception: {}".format(file_path, e))
+            s3_version = upload_response['VersionId']
+            path = self._get_path() + "/uploads/{}/complete".format(upload_id)
+            self.session.put_resource(path=path, json={'s3_version': s3_version})
+
+        file_link_dict = {
+            'filename': dest_name,
+            'url': self._get_path(object_key)
```

Comment:
what is this url set as?
is it `api/v1/projects/:id/datasets/:id/files/:id`?

Reply:
There's no `api/v1`. That part is appended by the Session object

Reply:
But otherwise, yes